projects/stargazer/plugins/other/radius/radius.cpp

```diff
@@ -6,26 +6,110 @@
 
 #include <boost/asio.hpp>
 #include <string>
+#include <iterator>
 #include <iostream>
-#include <cstdint> //uint8_t, uint32_t
 
 using STG::RADIUS;
+using STG::RAD_SETTINGS;
 
 extern "C" STG::Plugin* GetPlugin()
 {
     static RADIUS plugin;
     return &plugin;
 }
 
-std::string RADIUS::GetVersion() const
+RAD_SETTINGS::RAD_SETTINGS()
+    : m_port(0)
+{}
+
+int RAD_SETTINGS::ParseSettings(const ModuleSettings & s)
 {
-    return "Radius v.1.0";
+    ParamValue pv;
+    int p;
+
+    pv.param = "Port";
+    auto pvi = std::find(s.moduleParams.begin(), s.moduleParams.end(), pv);
+    if (pvi == s.moduleParams.end() || pvi->value.empty())
+    {
+        m_errorStr = "Parameter \'Port\' not found.";
```

Comment:
When "Port" is not found - it is not an error. This parameter is optional and we simply stick to the default value.